runtime/sema/check_assignment.go

```diff
@@ -212,6 +212,22 @@ func (checker *Checker) visitIndexExpressionAssignment(
 
 	elementType = checker.visitIndexExpression(target, true)
 
+	targetExpression := target.TargetExpression
+	switch targetExpression := targetExpression.(type) {
+	case *ast.MemberExpression:
+		_, member, _ := checker.visitMember(targetExpression)
```

Comment:
Why do we need to visit the targetExpression again here? I guess `checker.visitIndexExpression(target, true)` at L#213 already recursively visited the `targetExpression` (Or is it not?)

My two worries are: (1) the performance, (2) Wouldn't it report certain errors twice?

Reply:
`visitMember` caches calls to it, so calling it again here is really just a way to look up the previously computed member data in the cache, it doesn't cause any additional execution and doesn't report any additional errors.

Reply:
@dsainati1 Could you please add a comment above this line just to point this out for future readers? I remembered it was caching, but will likely forget again